docs/application/dotnet/guides/nui/styling-controls-with-xaml.md

```diff
@@ -0,0 +1,194 @@
+
+```csharp
+Selector selector = new Selector<Color>()
+{
+  Normal = Color.Black,
+  Pressed = Color.Blue,
+  Disabled = Color.Red
+};
+```
+
+The equivalent XAML code is as follows:
+
+```xml
+<Selector x:TypeArguments="Color" Normal="Black" Pressed="Blue" Disabled="Red"/>
+```
+
+The selectors work when the control state is enabled in a `View`. By default, the control state is enabled for all controls in `Tizen.NUI.Components` and not for the controls in `Tizen.NUI.BaseComponents`, such as `View` and `TextLabel`. If you want to enable a control state for them, set the `EnableControlState` property to `true`:
+
+```xml
+<ViewStyle x:Key="viewColoredByTouch" EnableControlState="true">
+  <ViewStyle.BackgroundColor>
+      <Selector x:TypeArguments="Color" Normal="Black" Pressed="Blue" Disabled="Red"/>
+  </ViewStyle.BackgroundColor>
+</ViewStyle>
+```
+}
+
+### Load theme in NUI application
+
+To use the theme implemented in XAML, follow these steps: 
+
+1. Create a theme object, `Theme theme = new Theme("PATH")`, where PATH is a full path to the XAML file in a project.
+2. Use `ThemeManager` to apply theme as `ThemeManager.apply(theme)`
+3. Use style defined in the XAML file in NUI component by setting the StyleName parameter to `component.StyleName = xaml_style_name`
+
+The following example explains the implementation of the `TextLabel` NUI component:
+
+```csharp
+using Tizen.NUI; 
+
+namespace NUI_Theme
+{
+  class Program : NUIApplication
+  {
+    protected override void OnCreate()
+    {
+      // Load theme from the xaml file
+      Theme theme = new Theme("full_path_of_the_xaml_file.xaml");
+
+      // Apply it to the current NUI Application
+      ThemeManager.ApplyTheme(theme);
+
+      // Set style name in NUI widget
+      TextLabel textLabel = new TextLabel()
+      textLabel.StyleName = "redTextStyle";
+    }
+  }
+```
+
+### Handle theme changes in runtime
+
+The following methods can be used to handle theme changes in runtime:
+
+- `ThemeManager.ThemeChanged`
+- `View.OnThemeChanged`
+* `ThemeManager.ThemeChanged`
+* `View.OnThemeChanged`
+
+When `ThemeManager.applyTheme` is called, `ThemeManager` calls the `ThemeChanged` callback. The following example shows how to handle `ThemeChanged` events. 
+
+```csharp
+using Tizen.NUI;
+
+namespace NUI_Theme
+{
+  class Program : NUIApplication
+  {
+    protected override void OnCreate()
+    {
+      //application init
+
+      ThemeManager.ThemeChanged += OnThemeChanged;
+    }
+
+    private void OnThemeChanged(object sender, ThemeChangedEventArgs args) 
+    {
+      //Implement reaction on theme change here.
+    }
+  }
+}
+```
+
+The `View.OnThemeChanged` should be used when creating a class derived from `View`. You can overload `OnThemeChanged` so that all functionalities related to the looks of the widget are implemented in one class:
```

Comment:
```suggestion
The `View.OnThemeChanged` is used when you create a class that is derived from `View`. You can overload `OnThemeChanged` so that all functionalities related to the looks of the widget are implemented in one class: